### PR TITLE
drop union return type from getaddrinfo

### PR DESCRIPTION
**Pull request by JelleZijlstra**
As discussed in #958. I added references to the related mypy issue, python/mypy#2509.

### CODE REVIEW

File: stdlib/2/socket.pyi [hunk 1]
```diff
@@ -335,10 +335,12 @@ def create_connection(address: Tuple[str, int],
                       source_address: Tuple[str, int] = ...) -> socket: ...
 
 # the 5th tuple item is an address
+# TODO the "Tuple[Any]" should be "Union[Tuple[str, int], Tuple[str, int, int, int]]" but that triggers
+# https://github.com/python/mypy/issues/2509
 def getaddrinfo(
         host: Optional[str], port: Union[str, int, None], family: int = ...,
         socktype: int = ..., proto: int = ...,
-        flags: int = ...) -> List[Tuple[int, int, int, str, Union[Tuple[str, int], Tuple[str, int, int, int]]]]:
+        flags: int = ...) -> List[Tuple[int, int, int, str, Tuple[Any]]]:
```
Comment (ambv) [inline stdlib/2/socket.pyi]: It should rather be `Tuple[Any, ...]`.

File: stdlib/3.4/asyncio/events.pyi [hunk 2]
```diff
@@ -93,8 +93,10 @@ class AbstractEventLoop(metaclass=ABCMeta):
     # Network I/O methods returning Futures.
     @abstractmethod
     @coroutine
+    # TODO the "Tuple[Any]" should be "Union[Tuple[str, int], Tuple[str, int, int, int]]" but that triggers
+    # https://github.com/python/mypy/issues/2509
     def getaddrinfo(self, host: str, port: int, *,
-        family: int = ..., type: int = ..., proto: int = ..., flags: int = ...) -> Generator[Any, None, List[Tuple[int, int, int, str, Union[Tuple[str, int], Tuple[str, int, int, int]]]]]: ...
+        family: int = ..., type: int = ..., proto: int = ..., flags: int = ...) -> Generator[Any, None, List[Tuple[int, int, int, str, Tuple[Any]]]]: ...
```
Comment (ambv) [inline stdlib/3.4/asyncio/events.pyi]: Ditto.

File: stdlib/3/socket.pyi [hunk 3]
```diff
@@ -331,10 +331,12 @@ def create_connection(address: Tuple[str, int],
                       source_address: Tuple[str, int] = ...) -> socket: ...
 
 # the 5th tuple item is an address
+# TODO the "Tuple[Any]" should be "Union[Tuple[str, int], Tuple[str, int, int, int]]" but that triggers
+# https://github.com/python/mypy/issues/2509
 def getaddrinfo(
         host: Optional[str], port: Union[str, int, None], family: int = ...,
         socktype: int = ..., proto: int = ...,
-        flags: int = ...) -> List[Tuple[int, int, int, str, Union[Tuple[str, int], Tuple[str, int, int, int]]]]:
+        flags: int = ...) -> List[Tuple[int, int, int, str, Tuple[Any]]]:
```
Comment (ambv) [inline stdlib/3/socket.pyi]: Ditto.

**Comment by gvanrossum** [comment]
Still breaks...
```
$ mypy test-data/samples/crawl.py
test-data/samples/crawl.py:147: error: Need more than 1 value to unpack (2 expected)
```

**Comment by ambv** [comment]
> Still breaks...
With `Tuple[Any, ...]` it won't.

**Comment by gvanrossum** [comment]
Try again with three dots. Then sit back, relax, wait three minutes, run the test against crawl.py and crawl2.py, *then* push. :-)

**Pull request merged**